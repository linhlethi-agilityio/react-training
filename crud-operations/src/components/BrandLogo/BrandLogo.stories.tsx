import type { Meta, StoryObj } from '@storybook/react';

// Components
import BrandLogo from '.';

const meta: Meta<typeof BrandLogo> = {
  component: BrandLogo,
};

export default meta;
type Story = StoryObj<typeof BrandLogo>;

export const Primary: Story = {
  args: {
    isAcronym: false,
    onClick: () => null,
  },
};

export const LgBrandLogo: Story = {
  args: {
    size: 'lg',
    isAcronym: true,
    onClick: () => null,
  },
};
