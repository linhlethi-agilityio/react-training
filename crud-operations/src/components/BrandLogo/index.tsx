import { memo } from 'react';
import { Heading, Link } from '@chakra-ui/react';

// Constants
import { BRAND_NAME } from '@constants';

// Utils
import { acronymText } from '@utils';

interface BrandLogoProps {
  size?: 'sm' | 'lg';
  isAcronym?: boolean;
  onClick: () => void;
}

const BrandLogo = ({ size = 'sm', isAcronym, onClick }: BrandLogoProps) => (
  <Link
    as="button"
    w="fit-content"
    _hover={{
      textDecoration: 'none',
    }}
    data-testid="brand-logo"
    onClick={onClick}
  >
    <Heading
      size={size}
      paddingLeft={3}
      textTransform="uppercase"
      borderLeftWidth={4}
      borderLeftColor="border.secondary"
    >
      {isAcronym ? acronymText(BRAND_NAME) : BRAND_NAME}
    </Heading>
  </Link>
);

export default memo(BrandLogo);
