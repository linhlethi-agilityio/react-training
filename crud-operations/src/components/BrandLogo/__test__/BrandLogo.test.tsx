import { render } from '@test-utils';
import '@testing-library/jest-dom';

// Components
import BrandLogo from '..';

const mockProps = {
  onClick: jest.fn(),
};

describe('BrandLogo component', () => {
  test('should match snapshot for BrandLogo', () => {
    const { container } = render(<BrandLogo {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
