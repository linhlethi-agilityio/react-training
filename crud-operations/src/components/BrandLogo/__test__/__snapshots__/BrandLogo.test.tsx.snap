// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogo component should match snapshot for BrandLogo 1`] = `
<div>
  <a
    class="chakra-link css-1mgm4dv"
  >
    <h2
      class="chakra-heading css-qis5cd"
    />
  </a>
</div>
`;
